cli/tools/repl/editor.rs

```diff
@@ -389,25 +396,35 @@ impl ReplEditor {
       EventHandler::Conditional(Box::new(TabEventHandler)),
     );
 
-    ReplEditor {
+    let history_file_dir = history_file_path.parent().unwrap();
+    std::fs::create_dir_all(history_file_dir).with_context(|| {
+      format!(
+        "Unable to create directory for the history file: {}",
+        history_file_dir.display()
+      )
+    })?;
+
+    Ok(ReplEditor {
       inner: Arc::new(Mutex::new(editor)),
       history_file_path,
-    }
+      errored_on_history_save: Arc::new(AtomicBool::new(false)),
+    })
   }
 
   pub fn readline(&self) -> Result<String, ReadlineError> {
     self.inner.lock().readline("> ")
   }
 
-  pub fn add_history_entry(&self, entry: String) {
+  pub fn update_history(&self, entry: String) {
     self.inner.lock().add_history_entry(entry);
-  }
-
-  pub fn save_history(&self) -> Result<(), AnyError> {
-    std::fs::create_dir_all(self.history_file_path.parent().unwrap())?;
+    if let Err(e) = self.inner.lock().save_history(&self.history_file_path) {
```

Comment:
What would be better is if this worked for multiple repls running at the same time.

Reply:
Can you elaborate?

Reply:
1. Open two repls
2. Execute `1;` in the first.
3. Execute `2;` in the second.
4. Close both repls.
5. Open a new repl and look at the history.

Reply:
It might be possible using [`append_history`](https://docs.rs/rustyline/latest/rustyline/struct.Editor.html#method.append_history). I'll give it a try

Reply:
Yeah, that does it.

Reply:
Cool